libwallet/src/internal/scan.rs

```diff
@@ -142,6 +148,88 @@ where
 	Ok(wallet_outputs)
 }
 
+fn collect_chain_outputs_view_wallet<'a, C>(
+	client: C,
+	rewind_hash: String,
+	start_index: u64,
+	end_index: Option<u64>,
+	status_send_channel: &Option<Sender<StatusMessage>>,
+) -> Result<ViewWallet, Error>
+where
+	C: NodeClient + 'a,
+{
+	let batch_size = 1000;
+	let start_index_stat = start_index;
+	let mut start_index = start_index;
+	let mut res = ViewWallet {
+		rewind_hash: rewind_hash,
+		output_result: vec![],
+		total_balance: 0,
+		last_pmmr_index: 0,
+	};
+	let secp = Secp256k1::with_caps(ContextFlag::VerifyOnly);
+
+	loop {
+		let (highest_index, last_retrieved_index, outputs) =
+			client.get_outputs_by_pmmr_index(start_index, end_index, batch_size)?;
+
+		let range = highest_index as f64 - start_index_stat as f64;
+		let progress = last_retrieved_index as f64 - start_index_stat as f64;
+		let percentage_complete = cmp::min(((progress / range) * 100.0) as u8, 99);
+
+		let msg = format!(
+			"Checking {} outputs, up to index {}. (Highest index: {})",
+			outputs.len(),
+			highest_index,
+			last_retrieved_index,
+		);
+		if let Some(ref s) = status_send_channel {
+			let _ = s.send(StatusMessage::Scanning(msg, percentage_complete));
+		}
+
+		// Scanning outputs
+		for output in outputs.iter() {
+			let (commit, proof, is_coinbase, height, mmr_index) = output;
+			let rewind_hash = from_hex(res.rewind_hash.as_str())
+				.map_err(|e| ErrorKind::GenericError(format!("Unable to create nonce: {}", e)))?;
+			let rewind_nonce = blake2b(32, &commit.0, &rewind_hash);
```

Comment:
https://github.com/mimblewimble/grin-wallet/issues/105 describes the hash as `rewind_nonce = H(H(pub_root_key), commit)` while you have in the code `rewind_nonce = H(commit, H(pub_root_key))`. I have yet to find whether the implementation actually switches these two around. If you happen to know where I can find this, let me know.

Reply:
I'm not sure exactly as of right now which one is true. I'm gonna look at the blake rfc more in depth. Will keep you updated.

FYI, I've just looked to the following function https://github.com/mimblewimble/grin/blob/master/core/src/libtx/proof.rs#L360-L364